server/src/main/java/org/elasticsearch/index/mapper/MappingLookup.java

```diff
@@ -291,15 +291,12 @@ private void checkObjectDepthLimit(long limit) {
     }
 
     private void checkFieldNameLengthLimit(long limit) {
-        Stream.of(objectMappers.values().stream(), fieldMappers.values().stream())
-            .reduce(Stream::concat)
-            .orElseGet(Stream::empty)
-            .forEach(mapper -> {
-                String name = mapper.simpleName();
-                if (name.length() > limit) {
-                    throw new IllegalArgumentException("Field name [" + name + "] is longer than the limit of [" + limit + "] characters");
-                }
-            });
+        Stream.concat(objectMappers.values().stream(), fieldMappers.values().stream()).forEach(mapper -> {
```

Comment:
Could we just solve this with a normal loop (or two)? The stream stuff is just needless overhead here I think.

Reply:
Yeah, good point